### PR TITLE
Log level/type: Use enum class instead of the awkward namespace trick.

### PR DESCRIPTION
**Pull request by hrydgard**
Just a small cleanup I've wanted to do for a long time. Good to do before making changes.

### CODE REVIEW

**Comment by fp64** [comment]
Offtopic, but while on subject of logs, is printing only file+line (not file+function+line) by design?

**Comment by hrydgard** [comment]
Yes, function names don't add that much value and makes the log lines longer and harder to read.

**Comment by hrydgard** [comment]
I'm merging and hoping for whatever CI cache to clean itself up. The error message is clearly not about the current state of the repo.

File: Common/LogManager.cpp [hunk 1]
```diff
@@ -69,64 +69,61 @@ void GenericLog(LogTypes::LOG_LEVELS level, LogTypes::LOG_TYPE type, const char
...
 };
 
 LogManager::LogManager(bool *enabledSetting) {
 	g_bLogEnabledSetting = enabledSetting;
 
-	for (size_t i = 0; i < ARRAY_SIZE(logTable); i++) {
-		_assert_msg_(i == logTable[i].logType, "Bad logtable at %i", (int)i);
-		truncate_cpy(log_[logTable[i].logType].m_shortName, logTable[i].name);
-		log_[logTable[i].logType].enabled = true;
+	_dbg_assert_(ARRAY_SIZE(g_logTypeNames) == (size_t)LogType::NUMBER_OF_LOGS);
```
Comment (unknownbrackets) [inline Common/LogManager.cpp:118]: Hm, couldn't we do a `static_assert` instead?

-[Unknown]

**Pull request merged**